### PR TITLE
Kernel check during runtime

### LINKED ISSUE #128: Kernel version during runtime
We need to bring the kernel check inside the collector using the shared library, this will help us to keep this repository.

### PR DESCRIPTION
**Pull request by thiagoftsm**
Fixes #128 

This PR brings the necessary changes for we reduce the number of libraries compiled.

### CODE REVIEW

File: library/bpf_load.c [hunk 1]
```diff
@@ -42,6 +42,45 @@ int prog_array_fd = -1;
...
+        version++;
+    else
+        return 0;
+
+    move = patch;
+    while (*version) *move++ = *version++;
+    *move = '\0';
+
+    return ((int)(atoi(major)*65536) + (int)(atoi(minor)*256) + (int)atoi(patch));
+}
```
Comment (mfundul) [inline library/bpf_load.c:82]: Why not reuse https://github.com/netdata/kernel-collector/blob/738dd982cf942cbdbf8489fbe9243af463924e93/libbpf/src/libbpf.c#L601 ?

Comment (thiagoftsm) [reply library/bpf_load.c]: It was because I did not know this function was inside `libbpf`, I only used the `bpf_` function from the API.
Thank you!

**Comment by thiagoftsm** [comment]
@mfundul I decided do not change the `libbpf` so I brought back the function.

**Comment by mfundul** [comment]
> 
> 
> @mfundul I decided do not change the `libbpf` so I brought back the function.

Ok but why don't you copy-paste the `libbpf` one instead of adding a new parser that we have to debug?

**Comment by thiagoftsm** [comment]
> > @mfundul I decided do not change the `libbpf` so I brought back the function.
> 
> Ok but why don't you copy-paste the `libbpf` one instead of adding a new parser that we have to debug?

Right, I will add here, because we have all the headers to compile it in place, but the version that I was using would be better to avoid necessity to install packages in few distributions.

**Comment by mfundul** [comment]
> 
> 
> > > @mfundul I decided do not change the `libbpf` so I brought back the function.
> > 
> > 
> > Ok but why don't you copy-paste the `libbpf` one instead of adding a new parser that we have to debug?
> 
> Right, I will add here, because we have all the headers to compile it in place, but the version that I was using would be better to avoid necessity to install packages in few distributions.

If you don't want to use `uname()` you could simply copy the rest of the code:
```
sscanf(info.release, "%u.%u.%u", &major, &minor, &patch) != 3
```

You don't need to add 30 lines of parsing code.

**Comment by thiagoftsm** [comment]
> > > > @mfundul I decided do not change the `libbpf` so I brought back the function.
> > > 
> > > 
> > > Ok but why don't you copy-paste the `libbpf` one instead of adding a new parser that we have to debug?
> > 
> > 
> > Right, I will add here, because we have all the headers to compile it in place, but the version that I was using would be better to avoid necessity to install packages in few distributions.
> 
> If you don't want to use `uname()` you could simply copy the rest of the code:
> 
> ```
> sscanf(info.release, "%u.%u.%u", &major, &minor, &patch) != 3
> ```
> 
> You don't need to add 30 lines of parsing code.

As I said, for this environment is not a problem, I will copy everything, but for `netdata/netdata` I will copy only what is really necessary.

**Pull request merged**